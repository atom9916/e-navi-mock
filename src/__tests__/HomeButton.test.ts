/**
 * @jest-environment jsdom
 */

import { render, screen } from '@testing-library/vue'
import HomeButtonVue from '@/components/HomeButton.vue'
import { createRouter, createWebHistory } from 'vue-router'
import { mount } from '@vue/test-utils'

describe('HomeButtonコンポーネントのテスト', () => {
  test('ボタンが正常に表示される', () => {
    render(HomeButtonVue)
    const button = screen.getByRole('button')
    expect(button).toHaveTextContent('ホーム')
  })
  test('ボタンをクリックするとホーム画面に遷移する', async () => {
    const router = createRouter({
      history: createWebHistory(),
      routes: [
        {
          path: '/',
          component: { template: '<div>Home</div>' }
        }
      ]
    })

    const wrapper = mount(HomeButtonVue, {
      global: {
        plugins: [router]
      }
    })

    await wrapper.find('button').trigger('click')

    expect(wrapper.vm.$route.path).toBe('/')
  })
  test('スナップショット', () => {
    const { container } = render(HomeButtonVue)
    expect(container).toMatchSnapshot()
  })
})
