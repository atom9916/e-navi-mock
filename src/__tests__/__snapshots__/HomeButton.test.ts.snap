// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeButtonコンポーネントのテスト スナップショット 1`] = `
<div>
  <div>
    <button>
      ホーム
    </button>
  </div>
</div>
`;
